nodes/viz/texture_viewer.py

```diff
@@ -0,0 +1,286 @@
+            bgl.glVertex2f(*coord)
+
+        bgl.glEnd()
+
+    def draw_texture(x=0, y=0, w=30, h=10, texname=texname):
+        #function to draw a texture
+        bgl.glEnable(bgl.GL_TEXTURE_2D)
+        bgl.glTexEnvf(bgl.GL_TEXTURE_ENV, bgl.GL_TEXTURE_ENV_MODE, bgl.GL_REPLACE)
+        bgl.glBindTexture(bgl.GL_TEXTURE_2D, texname)
+
+        bgl.glBegin(bgl.GL_QUADS)
+
+        bgl.glTexCoord2d(0, 0); bgl.glVertex2f( x, y )
+        bgl.glTexCoord2d(1, 0); bgl.glVertex2f( x + w , y )
+        bgl.glTexCoord2d(1, 1); bgl.glVertex2f( x + w, y - h )
+        bgl.glTexCoord2d(0, 1); bgl.glVertex2f( x, y - h )
+
+        bgl.glEnd()
+
+        bgl.glDisable(bgl.GL_TEXTURE_2D)
+        #bgl.glDeleteTextures( 1, Buffer )
+        bgl.glFlush()
+
+    draw_texture(x=x, y=y, w=width, h=height, texname=texname)
+
+    draw_borders(x=x, y=y, w=width, h=height, color=grid_color)
+
+class SvTextureViewerNode(bpy.types.Node, SverchCustomTreeNode):
+    '''Texture Viewer node'''
+    bl_idname = 'SvTextureViewerNode'
+    bl_label = 'Texture viewer'
+
+    tex = []
+
+    n_id = StringProperty(default='')
+    activate = BoolProperty(
+        name='Show', description='Activate texture drawing',
+        default=True,
+        update=updateNode)
+
+    selected_mode = EnumProperty(
+        items=size_tex_list,
+        description="offers display sizing",
+        default="SMALL",
+        update=updateNode
+    )
+
+    bitmap_save = EnumProperty(
+        items=bitmap_save_list,
+        description="offers bitmap saving",
+        default="PNG"
+    )
+
+    directory = StringProperty(
+        maxlen=1024,
+        subtype='FILE_PATH',
+        options={'HIDDEN', 'SKIP_SAVE'}
+    )
+
+    in_float = FloatProperty(
+        min=0.0, max=1.0, default=0.0, name='Float Input',
+        description='input for texture', update=updateNode
+    )
+
+    theme_mode_options = [(m, m, '', idx) for idx, m in enumerate(["default", "scope"])]
+    selected_theme_mode = EnumProperty(
+        items=theme_mode_options, default="default", update=updateNode
+    )
+
+    def draw_buttons(self, context, l):
+        c = l.column()
+        c.label(text="set texture display")
+        c.prop(self, "selected_mode", text="")
+        c.prop(self, 'activate')
+
+    def draw_buttons_ext(self, context, l):
+        l.label(text="choose a different color for the border:")
+        l.prop(self, "selected_theme_mode")
+        l.separator()
+        l.label(text="save texture as bitmap image, choose a format:")
+        l.prop(self, "bitmap_save")
+        l.operator('file.select_all_toggle', text='choose a directory')
+
+        row = l.row()
+        #addon = context.user_preferences.addons.get(sverchok.__name__)
+        #row.scale_y = 4.0 if addon.preferences.over_sized_buttons else 1
+
+        l.operator("node.scriptlite_ui_callback", text="S A V E").fn_name="save_bitmap"
+
+    def invoke(self, context, event):
```

Comment:
Does this work? to have a node behave as an operator?
The callback we do something far simpler than a full operator